prio/util.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2018, Henry Corrigan-Gibbs
+ * 
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. 
+ */
+
+#ifndef __UTIL_H__
+#define __UTIL_H__
+
+#include <mprio.h>
+#include "mpi/mpi.h"
+
+// Minimum of two values
+#define MIN(a, b) ((a) < (b) ? (a) : (b))
+
+// Check a Prio error code and return failure if the call fails.
+#define P_CHECK(s) \
```

Comment:
Not a big fan of this style of macros for error checking. But nothing I'd change now.

Reply:
Yeah, this feels ugly to me too. Is there a cleaner design pattern that I should use in the future?

Reply:
Right. So I'll leave this as is for now.